app/javascript/extensible-components/index.ts

```diff
@@ -0,0 +1,67 @@
+import { IExtensionComponent, IMiQApiCallback } from './lib';
+
+/**
+ * Class for easy creation of extensible component.
+ */
+export class ExtensibleComponent {
+  public unsubscribe: Function;
+  constructor(public name: string, public api: IMiQApiCallback, public render: IMiQApiCallback){}
+}
+
+/**
+ * Create new object which will hold extension components on MiQ main object.
+ */
+ManageIQ.extensionComponents = ManageIQ.extensionComponents || {};
+
+/**
+ * Subject from from Rxjs to send message that we want to register new component.
+ */
+ManageIQ.extensionComponents.source = new Rx.Subject();
```

Comment:
I think the `source` is pretty similar to our Redux `rootReducer` so I don't think we should expose it to clients.

Instead, we should expose useful API methods like `addComponent` and `subscribe` that internally work with `source`.

Reply:
Well the is `source` works as event emiter, you'll pass newComponent to it and it will notify everyone that new component was added, items are just persistent object so when someone starts to listen on source later on, it can check which components has been registered and extend them.

Reply:
But the source can be local variable so no need to expose it to outside world.